src/main/groovy/org/pih/warehouse/core/date/InstantParser.groovy

```diff
@@ -0,0 +1,174 @@
+package org.pih.warehouse.core.date
+
+import java.time.DateTimeException
+import java.time.Instant
+import java.time.LocalDate
+import java.time.ZoneId
+import java.time.ZonedDateTime
+import org.apache.commons.lang.StringUtils
+import org.apache.poi.hssf.usermodel.HSSFWorkbook
+import org.apache.poi.ss.usermodel.DateUtil
+import org.apache.poi.xssf.usermodel.XSSFWorkbook
+import org.springframework.stereotype.Component
+
+import org.pih.warehouse.databinding.DataBindingConstants
+
+/**
+ * Deserializes input objects into Instants.
+ *
+ * For use only by data importers. APIs should always use command objects which automatically handle data binding.
+ */
+@Component
+class InstantParser extends AbstractDateParser<Instant> {
+
+    @Override
+    Instant parse(Object date, DateParserContext context=null) {
+        switch (date) {
+            case String:
+                return asInstant(date as String, currentTimezone, context)
+            case double:
+            case Double:
+            case BigDecimal:
+                return asInstant(date as double, currentTimezone, context)
+            case ZonedDateTime:
+                return asInstant(date as ZonedDateTime)
+            case LocalDate:
+                return asInstant(date as LocalDate, currentTimezone)
+            case Date:
+                return asInstant(date as Date)
+            case Calendar:
+                return asInstant(date as Calendar)
+            case null:
+                return null
+            default:
+                throw new UnsupportedOperationException("Cannot parse date of type [${date.class}]")
+        }
+    }
+
+    /**
+     * Null-safe conversion of a Double to an Instant.
+     * This is how Excel represents date fields (aka serial numbers) and so is useful when importing xlsx files.
+     *
+     * @param excelDate An Excel formatted double representing a date
+     * @param zone "Z", or "UTC" or "+01:00" or "America/Anchorage" for example.
+     */
+    static Instant asInstant(Double excelDate, ZoneId zone, DateParserContext context) {
```

Comment:
This logic is new. It allows us to handle excel dates which are doubles. It isn't used yet though. I'll use it in the CC refactor PR